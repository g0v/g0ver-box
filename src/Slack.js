import _ from 'lodash';
import slack from 'slack';

export const bot = slack.rtm.client();

const TOKEN = process.env.SLACK_TOKEN;

if (TOKEN) {
  bot.listen({ token: TOKEN });
}

bot.hello((message) => {
  console.log(`Got a message: ${message.type}`);
});

const call = (name, req = {}) => (
  new Promise((resolve, reject) => {
    const fn = _.get(slack, name);
    fn(
      { ...req, token: TOKEN },
      (err, res) => (err ? reject(err) : resolve(res))
    );
  })
);

export default {
  userInfo: async req => await call('users.info', req),
  postMessage: async req => await call('chat.postMessage', { ...req, as_user: true }),
  postMultiMessage: async (users, message) => _.map(users, user => bot.ws.send(JSON.stringify({
    id: `${user}::${Date.now()}`, type: 'message', channel: user, text: message,
  }))),
  channelInfo: async req => await call('channels.info', req),
  channelJoin: async req => await call('channels.join', req),
};
