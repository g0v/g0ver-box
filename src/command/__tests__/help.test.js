import Slack from '../../Slack';
import index from '../';

describe('help command', () => {
  const data = { channel: 'D100', name: 'yutin', text: 'help' };

  it('default', async () => {
    await index(data);
    expect(Slack.postMessage.mock.calls).toMatchSnapshot();
  });
});
