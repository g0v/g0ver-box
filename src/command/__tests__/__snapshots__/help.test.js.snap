// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`help command default 1`] = `
Array [
  Array [
    Object {
      "attachments": Array [
        Object {
          "color": "#000",
          "mrkdwn_in": Array [
            "text",
            "pretext",
            "fields",
          ],
          "text": "*關鍵字 技能 議題*
\`add\`, \`del\`, \`whoami\`
\`add <hashtag>\` 新增關鍵字，逗號可以多筆 ( ex. add A, B )
\`del <hashtag>\` 刪除關鍵字，逗號可以多筆 ( ex. del A, B )
\`whoami\` 查詢 自己 登錄了哪些關鍵字",
        },
        Object {
          "color": "#000",
          "mrkdwn_in": Array [
            "text",
            "pretext",
            "fields",
          ],
          "text": "*推坑 找人*
\`whois\`, \`search\`
\`whois <@username>\` 查詢 g0ver 登錄了哪些關鍵字 ( ex. whois @yutin )
\`search <hashtag>\` *搜尋哪些 g0ver 有此關鍵字*",
        },
        Object {
          "color": "#000",
          "mrkdwn_in": Array [
            "text",
            "pretext",
            "fields",
          ],
          "text": "*入坑 出坑*
\`in\`, \`all\`, \`out\`
\`in <project / task> <hours>\` 進入某個坑裡，hours ( 選填 ) 預計幾小時後出坑
\`out\` 離開某個坑
\`all\` 顯示 g0ver 正在哪些坑，也許他們需要你的加入",
        },
        Object {
          "color": "#000",
          "mrkdwn_in": Array [
            "text",
            "pretext",
            "fields",
          ],
          "text": "*揪松 找松*
 To Do",
        },
        Object {
          "color": "#000",
          "mrkdwn_in": Array [
            "text",
            "pretext",
            "fields",
          ],
          "text": "*座右銘*
\`slogan\`
\`slogan <text>\` 持續關注的議題、來自哪裡 或 URL 讓大家認識你",
        },
        Object {
          "color": "#000",
          "mrkdwn_in": Array [
            "text",
            "pretext",
            "fields",
          ],
          "text": "*了解更多*
 <http://g0v.tw/zh-TW/manifesto.html|g0v 宣言> - <https://g0v-jothon.kktix.cc/|g0v 揪松團> - <https://github.com/g0v|g0v Github>",
        },
      ],
      "channel": "D100",
      "text": "歡迎來到 g0v。 社群是自主參與，成果開放
私訊 <@g0ver> 使用下列指令，當需要 *技能* 時可以找到沒有人，也可以發現 *適合入的坑* ，或者輸入 \`search g0v大使\` 這些人都願意推你入坑。",
    },
  ],
]
`;
